src/server/internalServer.cpp

```diff
@@ -602,12 +603,20 @@ std::unique_ptr<Response> InternalServer::handle_search(const RequestContext& re
     } catch(std::runtime_error& e) {
       // Searcher->search will throw a runtime error if there is no valid xapian database to do the search.
       // (in case of zim file not containing a index)
-      return HTTPErrorHtmlResponse(*this, request, MHD_HTTP_NOT_FOUND,
-                                   "fulltext-search-unavailable",
-                                   "404-page-heading",
-                                   m_root + "/skin/search_results.css")
-           + nonParameterizedMessage("no-search-results")
-           + TaskbarInfo(searchInfo.bookName, bookId.empty()?nullptr:mp_library->getArchiveById(bookId).get());
+      auto response = std::move(
+          HTTPErrorHtmlResponse(*this, request, MHD_HTTP_NOT_FOUND,
+                                "fulltext-search-unavailable",
+                                "404-page-heading",
+                                m_root + "/skin/search_results.css")
+        + nonParameterizedMessage("no-search-results")
+      );
+      if(searchInfo.bookNames.size() == 1) {
+        auto bookName = *searchInfo.bookNames.begin();
+        auto bookId = mp_nameMapper->getIdForName(bookName);
+        return response + TaskbarInfo(bookName, mp_library->getArchiveById(bookId).get());
+      } else {
+        return response;
+      }
```

Comment:
I don't like this way of building responses conditionally. I believe a better way is to introduce `operator +=()`, so that the code looks like below:

```c++
      HTTPErrorHtmlResponse response(*this, request, MHD_HTTP_NOT_FOUND,
                                    "fulltext-search-unavailable",
                                    "404-page-heading",
                                    m_root + "/skin/search_results.css");
      response += nonParameterizedMessage("no-search-results");
      if(searchInfo.bookNames.size() == 1) {
        auto bookName = *searchInfo.bookNames.begin();
        auto bookId = mp_nameMapper->getIdForName(bookName);
        response += TaskbarInfo(bookName, mp_library->getArchiveById(bookId).get());
      }
      return response;
```